url/src/host.rs

```diff
@@ -82,7 +82,12 @@ impl Host<String> {
             return parse_ipv6addr(&input[1..input.len() - 1]).map(Host::Ipv6);
         }
         let domain = percent_decode(input.as_bytes()).decode_utf8_lossy();
+
+        #[cfg(feature = "idna")]
         let domain = idna::domain_to_ascii(&domain)?;
+        #[cfg(not(feature = "idna"))]
+        let domain = domain.to_string();
```

Comment:
Maybe this should panic if `domain` is not ASCII? I think the rest of the code expects `Host::Domain` to hold the normalized domain name, so this way we don't introduce weird incompatibilities.

Reply:
Maybe not panic, but return an error?
Not only if the domain is not ASCII, but also if the domain contains any `xn--` labels - which we can't know if they are correct or no.

Reply:
I happen to be a backend engineer also, panic is the last thing I expect a base lib to do, especially when it may handle external input. I disable the idna doesn't mean I want my service panic for idna input...

Reply:
panic should used when the program enter some bad state and not easy to recover. For this case, it has no problem to recover.